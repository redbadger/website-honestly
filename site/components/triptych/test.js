import React from 'react';
import { mount } from 'enzyme';

import Triptych from './index';

function createClientXY(x, y) {
  return { clientX: x, clientY: y };
}

export function createStartTouchEventObject({ x = 0, y = 0 }) {
  return { touches: [createClientXY(x, y)] };
}

export function createMoveTouchEventObject({ x = 0, y = 0 }) {
  return { preventDefault: () => {}, changedTouches: [createClientXY(x, y)] };
}

describe('pages/our-work/case-study/pride/triptych', () => {
  it('renders correctly', () => {
    const rendered = mount(<Triptych />);
    expect(rendered).toMatchSnapshot();
  });

  it('changes the active image to the left one if the first radio button is pressed', () => {
    const rendered = mount(<Triptych />);
    const firstRadio = rendered.find('input').at(0);
    firstRadio.simulate('change', { target: { value: '0' } });
    expect(rendered.state('activeImg')).toEqual(0);
  });

  it('changes the active image to the right one if swiped left', () => {
    const rendered = mount(<Triptych />);
    const imgWrapper = rendered.find('.wrapper').get(0);
    // https://github.com/facebook/react/issues/9809
    imgWrapper.ref.current.ontouchstart(createMoveTouchEventObject({ x: 20, y: 0 }));
    imgWrapper.ref.current.ontouchend(createMoveTouchEventObject({ x: 10, y: 0 }));
    expect(rendered.state('activeImg')).toEqual(2);
  });

  it('builds the correct class for an active image', () => {
    const cssClass = Triptych.activeClass(0, 0);
    expect(cssClass).toEqual('fakeControl fakeControlActive');
  });
});
