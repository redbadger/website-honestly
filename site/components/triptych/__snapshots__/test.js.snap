// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/our-work/case-study/pride/triptych renders correctly 1`] = `
<Triptych>
  <div>
    <div
      className="wrapper"
    >
      <picture>
        <img
          className="leftImg"
        />
      </picture>
      <picture>
        <img
          className="middleImg"
        />
      </picture>
      <picture>
        <img
          className="rightImg"
        />
      </picture>
    </div>
    <div>
      <form
        action=""
        className="controls"
      >
        <label
          key="0"
        >
          <span
            className="fakeControl "
          />
          <span
            className="controlsLabel"
          >
            First
          </span>
          <input
            className="controlsInput"
            name="image"
            onChange={[Function]}
            type="radio"
            value={0}
          />
        </label>
        <label
          key="1"
        >
          <span
            className="fakeControl fakeControlActive"
          />
          <span
            className="controlsLabel"
          >
            First
          </span>
          <input
            className="controlsInput"
            name="image"
            onChange={[Function]}
            type="radio"
            value={1}
          />
        </label>
        <label
          key="2"
        >
          <span
            className="fakeControl "
          />
          <span
            className="controlsLabel"
          >
            First
          </span>
          <input
            className="controlsInput"
            name="image"
            onChange={[Function]}
            type="radio"
            value={2}
          />
        </label>
      </form>
    </div>
  </div>
</Triptych>
`;
